docs/proposals/20231123-enhance-mid-tier-resource.md

```diff
@@ -0,0 +1,253 @@
+
+koordinator bind koord-priority and resource model, different priorities have different resource models.
+
+This proposal introduce Mid+LS and Mid+BE fill the gap in Prod+LS and Batch+BE
+meet the requirements of different types of tasks.
+
+### Goals
+
+- How to calculate and update the mid-tier resource amount of a node.
+- Clarification of Mid-tier cgroup and QoS
+- Clarification of Mid-tier suppression and eviction policy
+- Necessary optimization of the scheduler/descheduler
+
+### Non-goals
+
+- Replace Batch-tier resources
+- Add new QoS type
+
+## Proposal
+
+### User stories
+
+#### Story 1
+
+There are online service tasks with low priority whose performance requirements match those of Prod+LS, but which should not be suppressed but can tolerate being evicted, when machine usage increases.
+
+Mid+LS can handle them.
+
+#### Story 2
+
+There are resource-intensive tasks such as AI or stream computing, e.g. Apache Spark, that can consume a lot of resources. They require stable resources, can be suppressed and do not want to be evicted.
+
+Mid+BE can defeat them.
+
+### Prerequisites
+
+Must use koordinator node reservation if someone wants to use Mid+LS
+
+### Design principles
+
+**QoS**
+
+By default, the performance of same QoS should be similar, Prod+LS and Mid+LS should be *basically* the same, also suitable to Mid+BE and Batch+BE.
+
+The adaptation of the QoS policy must be specific. In general, if the QoS is the same and the priorities are different, there is no need to make a big difference, such as the priority of eviction, and some advanced configurations in memory reclamation, usually no configuration is required.
+However, there is no finer configuration granularity for CPU group identity, so no additional customization is required for the time being.
+In the future, advanced features such as core schedule and CPU idle need to be considered.
+
+Also, you can introduce a QoS level, such as MID between LS and BE to implement functions such as QoS detail customization.
+
+**resource oversale**
+
+Mid resource is calculated by
+```
+Allocatable[Mid] := min(Reclaimable[Mid], NodeAllocatable * thresholdRatio)
+```
+at the moment we need to change this to
+
+```
+Allocatable[Mid] := min(Reclaimable[Mid], NodeAllocatable * thresholdRatio) + Unallocated[Mid]
+Unallocated[Mid] = max(NodeAllocatable - Allocated[Prod], 0)
+```
+
+reference [mid-tier-overcommitment](https://koordinator.sh/docs/designs/node-prediction/#mid-tier-overcommitment)
+
+in the long term, non-overselling can be supported by Feature-Gate.
+```
+Unallocated * thresholdRatio
+```
+
+**native resource or extended resource**
+
+*native resource*:
+hijack node update, change `node.Status.allocatable`, mid pod also use native resource, in this situation, Mid is equivalent to a sub-priority in prod, resource quota need to make adaptive modification.
```

Comment:
I don’t quite understand the logic described in this paragraph. Why do we need to hijack node update?